hub/api/dataset.py

```diff
@@ -220,19 +220,29 @@ def __iter__(self):
             yield self[i]
 
     def _load_meta(self):
-        if dataset_exists(self.storage):
+        meta_key = get_dataset_meta_key()
+
+        if meta_key in self.storage:
+            # dataset exists
+
             logger.info(f"Hub Dataset {self.path} successfully loaded.")
-            self.meta = DatasetMeta.load(self.storage)
+            self.meta = self.storage.get_cachable(meta_key, DatasetMeta)
```

Comment:
Do we want this to be an `@property` so it is refreshed in the cache each time it's accessed?

Reply:
no because of the printing thing, otherwise i would say yea

Reply:
(logger.info right above)

Reply:
@mccrearyd that's no excuse. I worry that this will be a major performance issue down the line. The sooner the better.